ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1357,6 +1357,7 @@ public void dropTable(Table table, boolean ifPurge) throws HiveException {
     table.getTTable().setTxnId(txnId);
 
     dropTable(table.getTTable(), !tableWithSuffix, true, ifPurge);
+    AcidUtils.tryInvalidateDirCache(table);
```

Comment:
````
dropTable(table.getTTable(), !tableWithSuffix, true, ifPurge);
````
could throw an exception, maybe better move inside of it and handle in finally block

Reply:
Thanks for checking it. I've moved the invalidation to the finally block.
My initial reasoning behind the invalidate call placement was, that I only wanted to invalidate the cache, if I was sure that the delete went through. So no exception happened.